### PR TITLE
Doxygen fixes

### PR DESCRIPTION
**Pull request by tjhei**
- add doxygen search engine
- add missing @params
- fix layout of list
- improve some documentation

### CODE REVIEW

**Comment by tjhei** [comment]
too late for 1.2 but useful anyways.

File: include/aspect/simulator.h [hunk 1]
```diff
@@ -883,21 +883,37 @@ namespace aspect
...
+       * @code
+       * template <int dim>
+       * class Functor
+       * {
+       *   public:
+       *     // operator() will have @p in and @p out filled out if @p true
+       *     bool need_material_properties();
+       *
+       *     // called once at the beginning with the number of quadrature points
+       *     void setup(unsigned int q_points);
```
Comment (bangerth) [inline include/aspect/simulator.h]: mark the argument as const

File: include/aspect/simulator.h [hunk 2]
```diff
@@ -883,21 +883,37 @@ namespace aspect
...
-       * MaterialModelOutputs & out, FEValues<dim> & fe_values, const
-       * LinearAlgebra::BlockVector &solution, std::vector<double> & output)
+       * The functor @p fctr should implement the following:
+       * @code
+       * template <int dim>
+       * class Functor
+       * {
+       *   public:
+       *     // operator() will have @p in and @p out filled out if @p true
+       *     bool need_material_properties();
```
Comment (bangerth) [inline include/aspect/simulator.h]: should the function be const?

File: include/aspect/simulator.h [hunk 3]
```diff
@@ -883,21 +883,37 @@ namespace aspect
...
       void get_artificial_viscosity (Vector<float> &viscosity_per_cell) const;
 
       /**
        * Internal routine to compute the depth average of a certain quantitiy.
-       * The functor @p fctr should implement: 1. bool
-       * need_material_properties() 2. void setup(unsigned int q_points) 3.
-       * double operator()(const MaterialModelInputs & in, const
-       * MaterialModelOutputs & out, FEValues<dim> & fe_values, const
-       * LinearAlgebra::BlockVector &solution, std::vector<double> & output)
+       * The functor @p fctr should implement the following:
```
Comment (bangerth) [inline include/aspect/simulator.h]: I'm not quite sure what you're trying to say here. Is the intent to say something like
```
  The functor @p fctr must be an object of a user defined type that can be arbitrary but has to satisfy certain requirements. In essence, this class type needs to implement the following interface of member functions (possibly in addition to other member functions, but they will not be called by this current function):
```

File: include/aspect/simulator.h [hunk 4]
```diff
@@ -883,21 +883,37 @@ namespace aspect
...
+       *        FEValues<dim> &fe_values,
+       *        const LinearAlgebra::BlockVector &solution,
+       *        std::vector<double> &output);
+       * };
+       * @endcode
        *
        * @param values The output vector of depth averaged values. The
        * function takes the pre-existing size of this vector as the number of
        * depth slices.
+       * @param fctr instance of a class satisfying the template above.
```
Comment (bangerth) [inline include/aspect/simulator.h]: template -> signature

Comment (bangerth) [inline include/aspect/simulator.h]: Capitalize "instance"

**Comment by tjhei** [comment]
Good points. I was hoping I could sneak the changes by you without going into the implementation. :-)

Okay like this? Should I squash?

**Comment by bangerth** [comment]
Yea, looks good. Please squash, my preference is always to have each commit be reasonably self-contained to the degree possible, and not leave open ends for the next commit. (I think in your case, I'd use the "fixup" option instead of squash, but that's your choice. I've really come to like "fixup" to hide my mistakes ;-)

**Pull request merged**